kernel/libs/aster-bigtcp/src/socket/bound.rs

```diff
@@ -337,18 +344,35 @@ impl<E: Ext> TcpConnection<E> {
         remote_endpoint: IpEndpoint,
         option: &RawTcpOption,
         observer: E::TcpEventObserver,
-    ) -> Result<Self, (BoundPort<E>, smoltcp::socket::tcp::ConnectError)> {
+    ) -> Result<Self, (BoundPort<E>, ConnectError)> {
+        let Some(local_endpoint) = bound.endpoint() else {
+            return Err((bound, ConnectError::Unaddressable));
+        };
+
+        let iface = bound.iface().clone();
```

Comment:
I believe this `clone()` can be removed if we manually drop `interface`/`sockets` before returning `bound` as an error. https://github.com/asterinas/asterinas/pull/1690#discussion_r1886092438

Reply:
The problem is that `sockets` must live until the connection is inserted.

```rust
// `bound` is consumed when constructing `connection`
let connection = Self::new(bound, inner);
connection.0.update_next_poll_at_ms(PollAt::Now);
connection.init_observer(observer);
// We need to hold the lock of `sockets` until the connection is inserted.
// But the lifetime of `sockets` is restricted by `bound` if iface is not cloned....
let res = sockets.insert_connection(connection.inner().clone());
```

Reply:
OK, thanks for the explanation. Then we may have to keep it, otherwise we cannot get the code to compile.